modules/apps/commerce/commerce-initializer-util/src/main/java/com/liferay/commerce/initializer/util/CommerceShipmentGenerator.java

```diff
@@ -158,8 +160,11 @@ private void _generateCommerceShipmentItems(
 				_getRandomCommerceInventoryWarehouse(
 					commerceInventoryWarehouses);
 
+			BigDecimal commerceOrderItemQuantity =
+				commerceOrderItem.getQuantity();
+
 			int quantity =
-				commerceOrderItem.getQuantity() -
+				commerceOrderItemQuantity.intValue() -
```

Comment:
I'm afraid we're missing the story to convert shipped quantity to BigDecimal

Reply:
I thought to manage it in COMMERCE-11741